bin/supervise.sh

```diff
@@ -1,31 +1,44 @@
 #!/usr/bin/env bash
 
+timestamp()
+{
+    if [[ -z "$TIMESTAMP_MODE" ]]; then
+        if ! command -v date >/dev/null; then
+            echo 'supervise.sh requires `date`' >&2
+            exit 1
+        fi
+
+        if [[ "$(date +%3N)" =~ ^[0-9]{3}$ ]]; then
+            TIMESTAMP_MODE="date_ns"
+        elif command -v adjtimex >/dev/null && command -v awk >/dev/null; then
+            TIMESTAMP_MODE="adjtimex"
+        else
+            TIMESTAMP_MODE="date_s"
+        fi
```

Comment:
this seems reasonable to me.  the only missing bit I think is macOS which has neither ns date nor adjtimex.  but this is also meant to run in containers on a k8s cluster so macOS is not really worth targeting for _full_ support, just enough to be able to test locally I think.

Reply:
> I think falling back to seconds for macOS is good enough

agreed!